### PR TITLE
RichCommands: SearchUnindexedItems

### PR DESCRIPTION
**Pull request by ferrariofilippo**
**Resolved / Related Issues**
- [x] Were these changes approved in an issue or discussion with the project maintainers? In order to prevent extra work, feature requests and changes to the codebase must be approved before the pull request will be reviewed. This prevents extra work for the contributors and maintainers.
   Related #11481 

**Validation**
How did you test these changes?
- [x] Did you build the app and test your changes?
- [x] Did you check for accessibility? You can use Accessibility Insights for this.
- [ ] Did you remove any strings from the en-us resource file?
   - [ ] Did you search the solution to see if the string is still being used? 
- [x] Did you implement any design changes to an existing feature?
   - [x] Was this change approved?
- [x] Are there any other steps that were used to validate these changes?
   1. Open app
   2. Search something that is not indexed
   3. Click on `Search unindexed items`

### CODE REVIEW

**Comment by yaira2** [comment]
@ferrariofilippo can you split the PR to help have a clearer commit history?

**Comment by ferrariofilippo** [comment]
@yair100 I saw you told me to split this PR, but now I no longer see the comment (did you delete it?), should I still open a separate PR for PermanentDeletion?

**Comment by yaira2** [comment]
It's preferred but I don't want to be too picky.

**Comment by yaira2** [comment]
I'm not convinced this should have an action, it's likely to be confused between the setting and the button on the search page.

**Comment by ferrariofilippo** [comment]
> I'm not convinced this should have an action, it's likely to be confused between the setting and the button on the search page.

I think we could rename it, but we definitely need to add this action. Otherwise we would need to keep `IBaseLayoutCommandImplementation` just for this action.

**Review by 0x5bfa — DISMISSED** [review]
First introduction of rich commands in layout modes!

**Comment by yaira2** [comment]
@ferrariofilippo after giving this some thought, I decided that we should remap the button on the search to open the settings page where users can toggle the setting for themselves.

The settings dialog doesn't have support for deep linking to specific pages (not yet) so we're going to have to hold off on this.

File: src/Files.App/Strings/en-US/Resources.resw [hunk 1]
```diff
@@ -3178,6 +3178,9 @@
   <data name="ShowCheckboxesWhenSelectingItems" xml:space="preserve">
     <value>Show checkboxes when selecting items</value>
   </data>
+  <data name="SearchUnindexedItemsDescription" xml:space="preserve">
+    <value>Include unindexed items in search results</value>
```
Comment (yaira2) [inline src/Files.App/Strings/en-US/Resources.resw:3182]: What about "Search for unindexed items"?

**Review by yair100 — APPROVED** [review]
LGTM

**Review by yair100 — APPROVED** [review]
LGTM

**Review by yair100 — APPROVED** [review]
LGTM

**Pull request merged**